frontend/src/concepts/roleBinding/RoleBindingPermissionsTable.tsx

```diff
@@ -1,56 +1,68 @@
 import * as React from 'react';
 import { Table } from '~/components/table';
 import { RoleBindingKind } from '~/k8sTypes';
-import { ProjectDetailsContext } from '~/pages/projects/ProjectDetailsContext';
-import { deleteRoleBinding, generateRoleBindingProjectSharing, createRoleBinding } from '~/api';
-import ProjectSharingTableRow from './ProjectSharingTableRow';
-import { columnsProjectSharing } from './data';
-import { ProjectSharingRBType } from './types';
+import { deleteRoleBinding, generateRoleBindingPermissions, createRoleBinding } from '~/api';
+import RoleBindingPermissionsTableRow from './RoleBindingPermissionsTableRow';
+import { columnsRoleBindingPermissions } from './data';
+import { RoleBindingPermissionsRBType, RoleBindingPermissionsRoleType } from './types';
 import { firstSubject } from './utils';
-import ProjectSharingTableRowAdd from './ProjectSharingTableRowAdd';
+import RoleBindingPermissionsTableRowAdd from './RoleBindingPermissionsTableRowAdd';
 
-type ProjectSharingTableProps = {
-  type: ProjectSharingRBType;
+type RoleBindingPermissionsTableProps = {
+  type: RoleBindingPermissionsRBType;
+  projectName: string;
+  roleRef?: string;
+  labels?: { [key: string]: string };
+  defaultGroupName?: string;
   permissions: RoleBindingKind[];
+  permissionOptions: {
+    type: RoleBindingPermissionsRoleType;
+    description: string;
+  }[];
   isAdding: boolean;
   typeAhead?: string[];
   onDismissNewRow: () => void;
   onError: (error: Error) => void;
   refresh: () => void;
 };
 
-const ProjectSharingTable: React.FC<ProjectSharingTableProps> = ({
+const RoleBindingPermissionsTable: React.FC<RoleBindingPermissionsTableProps> = ({
   type,
+  projectName,
+  roleRef,
+  labels,
+  defaultGroupName,
   permissions,
+  permissionOptions,
   typeAhead,
   isAdding,
   onDismissNewRow,
   onError,
   refresh,
 }) => {
-  const { currentProject } = React.useContext(ProjectDetailsContext);
-
   const [editCell, setEditCell] = React.useState<string[]>([]);
-
   return (
     <Table
       variant="compact"
       data={permissions}
-      data-testid={`project-sharing-table ${type}`}
-      columns={columnsProjectSharing}
+      data-testid={`role-binding-table ${type}`}
+      columns={columnsRoleBindingPermissions}
       disableRowRenderSupport
       footerRow={() =>
         isAdding ? (
-          <ProjectSharingTableRowAdd
+          <RoleBindingPermissionsTableRowAdd
             key="add-permission-row"
             type={type}
+            permissionOptions={permissionOptions}
             typeAhead={typeAhead}
             onChange={(name, roleType) => {
-              const newRBObject = generateRoleBindingProjectSharing(
-                currentProject.metadata.name,
+              const newRBObject = generateRoleBindingPermissions(
+                projectName,
                 type,
                 name,
-                roleType,
+                roleRef || roleType,
+                'Role',
```

Comment:
Instead of passing in `'Role'` here you need to lift the `rbRoleRefKind` prop out with the others so you can pass in Role for MR and ClusterRole for projects. I think that's what's causing your test failure

Reply:
oh, gah.. lol i think i just did a project wide copy/paste regex on generateRoleBindingPermissions and didn't check where it was 🤦